### PR TITLE
Move testdata to root

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Go mod automatically ignores the /testdata directory. This change ensures that the rootfs for source/pci tests are not included in the module definition.

### CODE REVIEW

File: source/pci/pci_test.go [hunk 1]
```diff
@@ -232,7 +232,8 @@ func TestPciSource(t *testing.T) {
 	// Run test cases
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
-			hostpath.SysfsDir = hostpath.HostDir(filepath.Join(packagePath, "testdata", tc.rootfs, "sys"))
+			mockSysfsPath := filepath.Join(packagePath, "..", "..", "testdata", "source", "pci", tc.rootfs, "sys")
```
Comment (elezar) [inline source/pci/pci_test.go]: Is the package path used for anything else other than the testdata dir? should we just set that directly instead?

Comment (ArangoGutierrez) [reply source/pci/pci_test.go]: umm you maybe right, `packagePath` has no other use, so we could even trim the `init` func at the beginning of the file

Comment (ArangoGutierrez) [reply source/pci/pci_test.go]: Done

**Comment by elezar** [comment]
Thanks @ArangoGutierrez. I squashed your changes onto https://github.com/kubernetes-sigs/node-feature-discovery/pull/1921

**Pull request closed without merging**